tests/ModelContextProtocol.Tests/Server/McpServerResourceTests.cs

```diff
@@ -1,25 +1,29 @@
-﻿using Microsoft.Extensions.DependencyInjection;
+﻿using Microsoft.Extensions.AI;
+using Microsoft.Extensions.DependencyInjection;
 using ModelContextProtocol.Protocol.Types;
 using ModelContextProtocol.Server;
+using Moq;
+using System.Reflection;
+using System.Text.Json.Serialization;
 
 namespace ModelContextProtocol.Tests.Server;
 
-public class McpServerResourceTests
+public partial class McpServerResourceTests
```

Comment:
Do we want to add any tests similar to Can_Be_Notified_Of_Tool_Changes? The logic is copied, so I doubt there are any problems, but it might be worth covering.